source/validate.cpp

```diff
@@ -518,6 +436,109 @@ void DebugInstructionPass(ValidationState_t& _,
   }
 }
 
+// Allows for piping of multiple functions which return spv_result_t
+spv_result_t operator||(spv_result_t lhs, spv_result_t rhs) {
```

Comment:
Two more thoughts about this:
* I could see how bitwise-or operator| could be read as confusing, since bitwise-or'ing of the spv_result_t integer values doesn't make sense.  But I think that's a better tradeoff than using the operator|| since, in *my* mind anyway, the short-circuiting is a stronger signal than the confusion of bitwise or-ing.   Basically, no C operator fits exactly right.
* Using an operator overload means both the LHS and RHS are now function arguments.  And so the order of evaluation is not specified.  Since there is no certain strict ordering between them, they are concurrently evaluated.  (Depends on who you ask what "concurrent" means....) The issue I'm concerned about is that, depending on the implementation, a module that fails more than one kind of check can will have an indeterminate ordering of messages from the different checks.   Now, the tests are set up to check only one set of rules at a time, so it won't show up in the tests.  It just might confuse users who are on different implementations, trading validation messages back and forth.  Not a big deal, but the absolutely safest thing to do would be to force evaluation of one set of tests at time, by using separate statements.

I still recommend what I said yesterday: please switch to operator|.  Otherwise, coder's choice.